cleaner/cleaner_main.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import time
+from datetime import timedelta
+from cleaner.scripts import clean_docker_cache
+from cleaner.worker import Worker
+from cleaner.utils.logger import LoggerMixin
+from cleaner.utils import common
+
+
+class Cleaner(LoggerMixin):
+
+    def __init__(self):
+        self.workers = {}
+
+    def add_worker(self, key, worker):
+        if key not in self.workers:
+            self.workers[key] = worker
+        else:
+            self.logger.warn("worker with key %s already exists.", key)
+
+    def start(self):
+        for k, w in self.workers.items():
+            w.start()
+            self.logger.info("worker %s started.", k)
+
+    def terminate(self):
+        for k, w in self.workers.items():
+            try:
+                # terminate the worker and all its subprocesses
+                common.kill_process_tree(w.pid, 1, self.logger)
```

Comment:
>1 [](start = 48, length = 1)

Is 1 second enough? How about 5 seconds to give more graceful time to the processes being killed?